shared/modules/transaction.utils.test.js

```diff
@@ -0,0 +1,83 @@
+import { isEIP1559Transaction, isLegacyTransaction } from './transaction.utils';
+
+describe('Transaction.utils', function () {
+  describe('isEIP1559Transaction', function () {
+    it('should return true if both maxFeePerGas and maxPriorityFeePerGas are hex strings', () => {
+      expect(
+        isEIP1559Transaction({
+          txParams: { maxFeePerGas: '0x1', maxPriorityFeePerGas: '0x1' },
+        }),
+      ).toBe(true);
+    });
+
+    it('should return false if either maxFeePerGas and maxPriorityFeePerGas are non-hex strings', () => {
+      expect(
+        isEIP1559Transaction({
+          txParams: { maxFeePerGas: 0, maxPriorityFeePerGas: '0x1' },
+        }),
+      ).toBe(false);
+      expect(
+        isEIP1559Transaction({
+          txParams: { maxFeePerGas: '0x1', maxPriorityFeePerGas: 'fail' },
+        }),
+      ).toBe(false);
+    });
+
+    it('should return false if either maxFeePerGas or maxPriorityFeePerGas are not supplied', () => {
+      expect(
+        isEIP1559Transaction({
+          txParams: { maxPriorityFeePerGas: '0x1' },
+        }),
+      ).toBe(false);
+      expect(
+        isEIP1559Transaction({
+          txParams: { maxFeePerGas: '0x1' },
+        }),
+      ).toBe(false);
+    });
+  });
+
+  describe('isLegacyTransaction', function () {
+    it('should return true if both maxFeePerGas and maxPriorityFeePerGas are not supplied', () => {
```

Comment:
maybe 'should return true if maxFeePerGas, maxPriorityFeePerGas and gasPrice are not supplied'?

Reply:
reworded to 'return true if no gas related fields are supplied'